src/main/shell-session/local-shell-session/local-shell-session.injectable.ts

```diff
@@ -17,10 +19,12 @@ interface InstantiationParameter {
 const localShellSessionInjectable = getInjectable({
   instantiate: (di, { cluster, tabId, webSocket }: InstantiationParameter) => {
     const createKubectl = di.inject(createKubectlInjectable);
+    const localShellEnvModifiers = di.inject(terminalShellEnvModifiersInjectable);
 
     const kubectl = createKubectl(cluster.version);
+    const entity = catalogEntityRegistry.getById(cluster.id);
 
-    return new LocalShellSession(kubectl, webSocket, cluster, tabId);
+    return new LocalShellSession(localShellEnvModifiers.get(), entity, kubectl, webSocket, cluster, tabId);
```

Comment:
It would be even better if `terminalShellEnvModifiersInjectable` returned a function of the signature `(clusterId: ClusterId, env: Record<string, string>) => Record<string, string>`.

That way this code doesn't need to know about the type `ShellEnvContext`.

Reply:
Where does it need to know about `ShellEnvContext`? Do you mean `catalogEntityRegistry`?

Reply:
Yes I do mean `catalogEntityRegistry`. Also, it would make the use of it within `open` simpler since we could just call it unconditionally.

Reply:
I pushed another commit. Is it what you were thinking here?